pkg/cli/mirror/operator.go

```diff
@@ -102,33 +101,41 @@ func (o *OperatorOptions) run(ctx context.Context, cfg v1alpha2.ImageSetConfigur
 	defer reg.Destroy()
 
 	mmapping := image.TypedImageMapping{}
-	for _, ctlg := range cfg.Mirror.Operators {
-
-		ctlgRef, err := image.ParseReference(ctlg.Catalog)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing catalog: %v", err)
-		}
+	for i, ctlg := range cfg.Mirror.Operators {
+		// Exclude catalog images with "oci://" from renderDC and plan
+		// These catalogs will be handled with bulkImageMirror instead
+		// TODO: we should be able to use renderDC still for these images
+		// so that the catalog image (filtered) gets recreated (and written
+		// to disk) for the OCI catalogs as well
+		if image.IsFBCOCI(ctlg.Catalog) {
+			cfg.Mirror.Operators = append(cfg.Mirror.Operators[:i], cfg.Mirror.Operators[i+1:]...)
```

Comment:
I assume you're trying to update the v1alpha2.ImageSetConfiguration to remove the operator from the list of operators. This is going to panic... here's a test mimics what this code is doing, and it results in a panic:

```go
func TestExampleThatPanics(t *testing.T) {
	cfg := v1alpha2.ImageSetConfiguration{
		ImageSetConfigurationSpec: v1alpha2.ImageSetConfigurationSpec{
			Mirror: v1alpha2.Mirror{
				Operators: []v1alpha2.Operator{
					{
						Catalog: "oci://foo/bar",
					},
					{
						Catalog: "foo.io/bar",
					},
					{
						Catalog: "oci://aaa/bbb",
					},
				},
			},
		},
	}

	for i, ctlg := range cfg.Mirror.Operators {
		if image.IsFBCOCI(ctlg.Catalog) {
			cfg.Mirror.Operators = append(cfg.Mirror.Operators[:i], cfg.Mirror.Operators[i+1:]...)
		}
	}
	// you'll never get here because you'll get "panic: runtime error: slice bounds out of range [3:1]"
	assert.Len(t, cfg.Mirror.Operators, 1)
	require.Equal(t, "foo.io/bar", cfg.Mirror.Operators[0].Catalog)
}
```

You need to do something different like this example if you want to update that slice:

```go
func TestExampleThatWorks(t *testing.T) {
	cfg := v1alpha2.ImageSetConfiguration{
		ImageSetConfigurationSpec: v1alpha2.ImageSetConfigurationSpec{
			Mirror: v1alpha2.Mirror{
				Operators: []v1alpha2.Operator{
					{
						Catalog: "oci://foo/bar",
					},
					{
						Catalog: "foo.io/bar",
					},
					{
						Catalog: "oci://aaa/bbb",
					},
				},
			},
		},
	}
	filtered := cfg.Mirror.Operators[:0]
	for _, ctlg := range cfg.Mirror.Operators {
		if !image.IsFBCOCI(ctlg.Catalog) {
			filtered = append(filtered, ctlg)
		}
	}
	cfg.Mirror.Operators = filtered
	require.Len(t, cfg.Mirror.Operators, 1)
	require.Equal(t, "foo.io/bar", cfg.Mirror.Operators[0].Catalog)
}
```

See https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating for example of this (and potential garbage collection that might be necessary depending on the use case).

At any rate, I am not certain this is needed at all since the argument `cfg` that is passed into the function is not a pointer, you're operating on a copy, so any modifications you made are lost when the function returns.

Reply:
That's correct! thanks for spotting this.